soccer/planning/trajectory/Trajectory.hpp

```diff
@@ -0,0 +1,130 @@
+#pragma once
+
+#include <memory>
+#include <planning/MotionInstant.hpp>
+#include <planning/paths/InterpolatedPath.hpp>
+#include <planning/paths/Path.hpp>
+#include <utility>
+
+namespace Trajectory {
+using Planning::AngleInstant, Planning::MotionInstant, Planning::Path,
+    Planning::InterpolatedPath, Planning::RobotInstant;
+
+using AngleFn = std::function<AngleInstant(MotionInstant)>;
+
+/**
+ * \brief Represents a trajectory x(t), y(t), h(t) that is sent from the
+ * PlannerNode to the MotionControlNode.
+ *
+ * TEMP: For simplicity, this is just a angleFunction attached to a
+ * InterpolatedPath.
+ */
+class Trajectory {
+public:
+    /**
+     * \brief Default constructor so we can make an array of this.
+     */
+    Trajectory() = default;
+
+    /**
+     * \brief Constructor that takes in any path and a angle function,
+     * and "rasterizes" the path to be an InterpolatedPath
+     *
+     * @param path
+     * @param angle_function
+     * @param granularity The granularity in time with which to construct this
+     * trajectory
+     */
+    Trajectory(std::unique_ptr<Path> path,
+               std::optional<AngleFn> angle_function, double granularity)
+        : path_(rasterizePath(std::move(path), angle_function, granularity)) {}
+
+    /**
+     * @return Whether the trajectory contains a valid nonempty path
+     */
+    [[nodiscard]] inline bool hasPath() const {
+        return path_ != nullptr && !path_->empty();
+    }
+
+    /**
+     * @return The start time of the path
+     */
+    [[nodiscard]] inline RJ::Time startTime() const {
+        return path_->startTime();
```

Comment:
Since the trajectory has a function to delete/take the path, it's probably extra safe to add some null ptr checks around all these getters. Even if it's for the sole purpose of simplifying the debugging process while working with this class. Honestly, even just asserts work since there's no good way to fail gracefully

Reply:
Done